merkle-tree/src/merkle_tree.rs

```diff
@@ -139,6 +139,85 @@ impl MerkleTree {
         self.nodes.iter().last()
     }
 
+    fn append_nodes(nodes: &mut [Hash], mut leaf_count: usize, leaves: Vec<Hash>) -> usize {
+        for mut leaf in leaves {
+            let mut layer: usize = 0;
+            leaf_count += 1;
+            let mut cursor = leaf_count;
+            while (cursor & 1) == 0 {
+                let arg1 = &leaf;
+                let arg2 = &nodes[layer];
```

Comment:
What is the point of these `arg1` and `arg2` here?
Can we just use `&leaf` and `&nodes[layer]` below?